src/main/java/com/hermesworld/ais/galapagos/kafka/impl/ConnectedKafkaCluster.java

```diff
@@ -137,14 +138,12 @@ public String getId() {
         ConfigResource res = new ConfigResource(Type.TOPIC, topicName);
         Config config = new Config(configs);
 
-        return toCompletableFuture(
-                this.adminClient
-                        .createTopics(
-                                Collections.singleton(new NewTopic(topicName, topicCreateParams.getNumberOfPartitions(),
-                                        (short) topicCreateParams.getReplicationFactor())))
-                        .all()).thenCompose(
-                                o -> toCompletableFuture(
-                                        adminClient.alterConfigs(Collections.singletonMap(res, config)).all()));
+        // TODO numPartitions should be configurable for Galapagos metadata topics
+        NewTopic newTopic = new NewTopic(topicName, 3, (short) topicCreateParams.getReplicationFactor());
+        newTopic = newTopic.configs(Map.of(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT));
```

Comment:
Nope, this is also only correct for Galapagos metadata topics. You must set all configs here which are in `config`. Please again check - line for line - what this code does :-)